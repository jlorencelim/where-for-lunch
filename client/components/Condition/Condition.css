.filter {
  > span, > div {
    display: inline;
  }
}
