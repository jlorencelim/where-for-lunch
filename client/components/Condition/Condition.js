import _ from 'lodash';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Input from '../Input/Input';
import Checkbox from '../Checkbox/Checkbox';
import styles from './Condition.css';

class Condition extends PureComponent {
  static propTypes = {
    condition: PropTypes.object,
    onRadiusConditionChanged: PropTypes.func,
    onPriceConditionChanged: PropTypes.func,
    price: PropTypes.arrayOf(PropTypes.number),
  };

  handleOnBlurAction = (e) => {
    this.props.onRadiusConditionChanged(e.target.value);
  }

  onCheckboxValueChanged = (e) => {
    const { checked, value } = e.target;
    const { condition: { price } } = this.props;

    if (checked) {
      price.push(Number(value));
    } else {
      _.remove(price, element => element === Number(value));
    }

    this.props.onPriceConditionChanged(price);
  }

  renderPriceCheckboxes = () => (
    _.range(1, 5).map(checkbox => (
      <Checkbox
        key={checkbox}
        label={'$'.repeat(checkbox)}
        value={checkbox}
        onChange={this.onCheckboxValueChanged}
      />
    ))
  )

  render() {
    const { condition: { radius } } = this.props;
    return (
      <div className={styles.root}>
        <div className={styles.filter}>
          <span>radius:</span>
          <Input defaultValue={radius} onBlurAction={this.handleOnBlurAction} />
          <span>meters</span>
        </div>
        <div className={styles.filter}>
          <span>price:</span>
          { this.renderPriceCheckboxes() }
        </div>
      </div>
    );
  }
}

export default Condition;
