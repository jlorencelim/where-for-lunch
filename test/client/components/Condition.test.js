import React from 'react';
import Condition from 'client/components/Condition/Condition';
import renderer from 'react-test-renderer';

test('With Jest snapshot, Condition component renders default state', () => {
  const condition = {
    radius: 500,
    longitude: null,
    latitude: null,
    price: [],
  };
  const conditionComponent = renderer
    .create(<Condition condition={condition} />)
    .toJSON();

  expect(conditionComponent).toMatchSnapshot();
});

test('With Jest snapshot, Condition component renders with condition values', () => {
  const condition = {
    radius: 1000,
    longitude: 1.00,
    latitude: 1.00,
    price: [1, 2, 3, 4],
  };
  const conditionComponent = renderer
    .create(<Condition condition={condition} />)
    .toJSON();

  expect(conditionComponent).toMatchSnapshot();
});
