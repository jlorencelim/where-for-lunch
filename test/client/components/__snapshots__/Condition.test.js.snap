// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, Condition component renders default state 1`] = `
<div
  className="root"
>
  <div
    className="filter"
  >
    <span>
      radius:
    </span>
    <div
      className="root"
    >
      <input
        defaultValue={500}
        onBlur={[Function]}
        type="text"
      />
    </div>
    <span>
      meters
    </span>
  </div>
  <div
    className="filter"
  >
    <span>
      price:
    </span>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={1}
      />
      <label>
        $
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={2}
      />
      <label>
        $$
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={3}
      />
      <label>
        $$$
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={4}
      />
      <label>
        $$$$
      </label>
    </div>
  </div>
</div>
`;

exports[`With Jest snapshot, Condition component renders with condition values 1`] = `
<div
  className="root"
>
  <div
    className="filter"
  >
    <span>
      radius:
    </span>
    <div
      className="root"
    >
      <input
        defaultValue={1000}
        onBlur={[Function]}
        type="text"
      />
    </div>
    <span>
      meters
    </span>
  </div>
  <div
    className="filter"
  >
    <span>
      price:
    </span>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={1}
      />
      <label>
        $
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={2}
      />
      <label>
        $$
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={3}
      />
      <label>
        $$$
      </label>
    </div>
    <div
      className="root"
    >
      <input
        onChange={[Function]}
        type="checkbox"
        value={4}
      />
      <label>
        $$$$
      </label>
    </div>
  </div>
</div>
`;
